pkg/adapter/awsdynamodbsource/adapter.go

```diff
@@ -209,42 +202,24 @@ func (a *adapter) processLatestRecords(shardIterator *string) error {
 		return fmt.Errorf("failed to get records: %w", err)
 	}
 
-	var wg sync.WaitGroup
-	wg.Add(len(getRecordsOutput.Records))
-
 	for _, record := range getRecordsOutput.Records {
-		go func(record *dynamodbstreams.Record) {
-			if err := a.sendDynamoDBEvent(record); err != nil {
-				a.logger.Errorw("Failed to send CloudEvent", zap.Error(err))
-			}
-		}(record)
-		wg.Done()
+		if err := a.sendDynamoDBEvent(record); err != nil {
+			a.logger.Errorw("Failed to send CloudEvent", zap.Error(err))
+		}
 	}
 
-	wg.Wait()
-
 	return nil
 }
 
 func (a *adapter) sendDynamoDBEvent(record *dynamodbstreams.Record) error {
 	a.logger.Info("Processing record ID: " + *record.EventID)
 
-	data := &DynamoDBEvent{
-		AwsRegion:    record.AwsRegion,
-		Dynamodb:     record.Dynamodb,
-		EventID:      record.EventID,
-		EventName:    record.EventName,
-		EventSource:  record.EventSource,
-		EventVersion: record.EventVersion,
-		UserIdentity: record.UserIdentity,
-	}
-
 	event := cloudevents.NewEvent(cloudevents.VersionV1)
 	event.SetType(v1alpha1.AWSEventType(a.arn.Service, strings.ToLower(*record.EventName)))
-	event.SetSubject(asEventSubject(record))
+	event.SetSubject(*record.Dynamodb.SequenceNumber)
```

Comment:
I implemented that weird looking function because setting the sequence number as the subject wasn't very much aligned with the description of a "subject" from the CloudEvents API.

I'm not sure whether my approach is the best we could come up with (using the name of the modified key(s) as subject), but I believe we should stay away from cryptic numbers in "subject" attributes.

There is another issue opened to describe this kind of things and what people should expect from our events: what convention do we use for "source" or "subject" for example, or what extensions we set, if any.

Reply:
> we should stay away from cryptic numbers

agree. I'm totally okay with restoring subjects
I didn't find any description of "subject" field and thought that it could be an opportunity to simplify the code, especially considering that event body now contains full DynamoDB record including keys and we can get rid of additional calculations on our side and leave it to the receiver